addons/mrp/migrations/10.0.2.0/post-migration.py

```diff
@@ -278,37 +284,112 @@ def populate_stock_move_workorder_id(cr):
 
 def fill_stock_move_unit_factor(env):
     cr = env.cr
-    productions = env['mrp.production'].search([])
-    if openupgrade.column_exists(cr, 'mrp_production', 'qty_produced'):
-        cr.execute(
-            """
-            SELECT id
-            FROM mrp_production
-            WHERE qty_produced IS NULL
-            """
-        )
-        ids = [r[0] for r in cr.fetchall()]
-        productions = env['mrp.production'].browse(ids)
-    productions._get_produced_qty()
+    # to fix performence issue, we use sql to compute quantity_done of
+    # stock move and qty_produced of mrp_production
+    quantity_done_field = 'quantity_done'
+    if not openupgrade.column_exists(cr, 'stock_move', 'quantity_done'):
```

Comment:
Even if this field exists, I'm thinking it would be wrong to take custody of it. For your purposes, I would *always* create the new column using get_legacy_name.